src/sentry/dynamic_sampling/rules/helpers/prioritize_transactions.py

```diff
@@ -0,0 +1,37 @@
+from typing import Mapping, Tuple, cast
+
+from sentry.dynamic_sampling.rules.utils import get_redis_client_for_ds
+from sentry.utils import json
+
+
+def _get_cache_key(org_id: int, proj_id: int) -> str:
+    return f"ds::o:{org_id}:p:{proj_id}:pri_tran"
+
+
+def get_transactions_resampling_rates(
+    org_id: int, proj_id: int, default_rate: float
+) -> Tuple[Mapping[str, float], float]:
+    redis_client = get_redis_client_for_ds()
+    cache_key = _get_cache_key(org_id=org_id, proj_id=proj_id)
+    try:
+        serialised_val = redis_client.get(cache_key)
+        if serialised_val:
+            val = json.loads(serialised_val)
+            ret_val = cast(Tuple[Mapping[str, float], float], val)
+            return ret_val
+    except (TypeError, ValueError):
+        # for any error just get the default
```

Comment:
do we what to capture exceptions here by sentry?

Reply:
Don't remember exactly why I have both, but it only catches deserialisation errors.

Reply:
`sentry_sdk.capture_exception(e)`